src/client/js/CoreEditor.js

```diff
@@ -84,13 +84,35 @@ class CoreEditor extends React.Component {
         const {
             editorState
         } = this.state;
-        const selection = editorState.getSelection();
-        if (!selection.isCollapsed()) {
-            this.setState({
-                showURLInput: 1,
-                urlValue: ''
-            });
+        var title = '';
+        const selectionState = editorState.getSelection();
+        const blockKey = selectionState.getAnchorKey();
+        const contentBlocks = editorState.getCurrentContent().getBlocksAsArray();
+        var start = selectionState.getStartOffset();
+        var end = selectionState.getEndOffset();
+
+        for (var i = contentBlocks.length - 1; i >= 0; i--) {
+         
+
+            var mapIter = contentBlocks[i].values();
+            var key = mapIter.next().value;
+            var type = mapIter.next().value;
+            var text = mapIter.next().value;
+
+            if(key === blockKey){
+                title = text.slice(start, end);
+                console.log(title);
             }
+
+        };
+
```

Comment:
@muddaserahmed code style issue, for loop is bloated and hardly readable
can be much simplified
```
 contentBlocks.forEach((block) => {
            if(block.key === blockKey){
                title = block.text.slice(start, end);
                console.log("TITLE: ",title);
            }
        });
```